restli-client/src/main/java/com/linkedin/restli/client/ExecutionGroup.java

```diff
@@ -0,0 +1,167 @@
+/*
+   Copyright (c) 2021 LinkedIn Corp.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+package com.linkedin.restli.client;
+
+import com.linkedin.parseq.Engine;
+import com.linkedin.parseq.ParTask;
+import com.linkedin.parseq.Task;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+
+/**
+ * {@link ExecutionGroup} is used to group requests generated by FluentApis, so the batching support provided
```

Comment:
Document that this is not threadsafe and should not be shared across threads.

Reply:
Updated